test/utils/ansible-test/lib/manage_ci.py

```diff
@@ -0,0 +1,140 @@
+"""Access Ansible Core CI remote services."""
+
+from __future__ import absolute_import, print_function
+
+import pipes
+
+from time import sleep
+
+import lib.pytar
+
+from lib.util import (
+    SubprocessError,
+    ApplicationError,
+    run_command,
+)
+
+from lib.core_ci import (
+    AnsibleCoreCI,
+)
+
+from lib.ansible_util import (
+    ansible_environment,
+)
+
+
+class ManageWindowsCI(object):
+    """Manage access to a Windows instance provided by Ansible Core CI."""
+    def __init__(self, core_ci):
+        """
+        :type core_ci: AnsibleCoreCI
+        """
+        self.core_ci = core_ci
+
+    def wait(self):
+        """Wait for instance to respond to ansible ping."""
+        extra_vars = [
+            'ansible_connection=winrm',
+            'ansible_host=%s' % self.core_ci.connection.hostname,
+            'ansible_user=%s' % self.core_ci.connection.username,
+            'ansible_password=%s' % self.core_ci.connection.password,
+            'ansible_port=%s' % self.core_ci.connection.port,
+            'ansible_winrm_server_cert_validation=ignore',
+        ]
+
+        env = ansible_environment(self.core_ci.args)
+        cmd = ['ansible', '-m', 'win_ping', '-i', 'windows,', 'windows', '-e', ' '.join(extra_vars)]
+
+        for _ in range(1, 60):
+            try:
+                run_command(self.core_ci.args, cmd, env=env)
+                return
+            except SubprocessError:
+                sleep(10)
+                continue
+
+        raise ApplicationError('Timeout waiting for %s/%s instance %s.' %
+                               (self.core_ci.platform, self.core_ci.version, self.core_ci.instance_id))
+
+
+class ManagePosixCI(object):
+    """Manage access to a POSIX instance provided by Ansible Core CI."""
+    def __init__(self, core_ci):
+        """
+        :type core_ci: AnsibleCoreCI
+        """
+        self.core_ci = core_ci
+        self.ssh_args = ['-o', 'StrictHostKeyChecking=no', '-i', self.core_ci.ssh_key.key]
```

Comment:
Might be worth running ssh in non-interactive mode. Hopefully not an issue in Shippable, though may be useful for local runs.
`'-o', 'BatchMode=yes'`